workshop.ipynb

```diff
@@ -135,6 +218,18 @@
     "git commit -m \"Initial Commit\""
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 4.2 Command Breakdown\n",
+    "**git reset** - using this command without arguments will alow us to reset the staging area mening the files we just added using. **git add** will be removed but the working tree will stay the same. ***WARNING***, the reset command provides a lot more functionality which at times may even deleted changed you've made to the working tree and subsequently all the working you still haven't commited, just make sure you know what exactly you are doing before using it.\n",
+    "\n",
+    "**git add** - Adding was covered [here](#4.1-Command-Breakdown) with an option, now using an argument we will just add the file that matches the parameter\n",
+    "\n",
+    "**git commit** - As the name says we are commiting to a change we made to the repository, once you commit a state of the repository it will be registered and you can always restore it or check differences to the current state. Commits can be deleted but its not something you do on a regular basis as it's usualy easier to fix something rather than redoing everything else. The \"m\" lets you add a message to your commit, these should usualy follow a protocol that lets you quicky understand what's beed added removed or modified with that commit"
```

Comment:
"once you commit **a** state of the repository **it** will" -> "once you commit the state of the repository will"